lms/static/scripts/frontend_apps/components/BasicLtiLaunchApp.js

```diff
@@ -23,53 +25,109 @@ import Spinner from './Spinner';
  * @property {string} displayName - User's display name
  */
 
-const INITIAL_LTI_LAUNCH_STATE = {
-  // The current state of the screen.
-  // One of "fetching", "fetched-url", "authorizing" or "error".
-  //
-  // When the app is initially displayed it will use the Via URL if given
-  // or invoke the API callback to fetch the URL otherwise.
-  state: 'fetching-url',
-
-  // URL of assignment content. Set when state is "fetched-url".
-  contentUrl: null,
-
-  // Details of last error. Set when state is "error".
-  error: null,
-
-  // The action that resulted in an error.
-  failedAction: null,
-};
-
 /**
  * Application displayed when an assignment is launched if the LMS backend
  * is unable to directly render the content in an iframe. This happens when
  * the content URL needs to be fetched from a remote source (eg. the LMS's
  * file storage) first, which may require authorization from the user.
  */
-export default function BasicLtiLaunchApp() {
+export default function BasicLtiLaunchApp({ rpcServer }) {
   const {
     api: {
       authToken,
       // API callback to use to fetch the URL to show in the iframe. This is
       // needed if resolving the content URL involves potentially slow calls
       // to third party APIs (eg. the LMS's file storage).
       viaCallbackUrl,
+      // Sync API callback and data to asynchronously load the section groups
+      // to relay to the sidebar via RPC.
+      sync: apiSync,
     },
     grading,
     // Content URL to show in the iframe.
     viaUrl,
     canvas,
   } = useContext(Config);
 
-  const [ltiLaunchState, setLtiLaunchState] = useState({
-    ...INITIAL_LTI_LAUNCH_STATE,
-    state: viaCallbackUrl ? 'fetching-url' : 'fetched-url',
-    contentUrl: viaUrl ? viaUrl : null,
-  });
+  // The current state of the error.
+  // One "error-fetch", "error-authorizing", or "error-report-submission"
+  const [errorState, setErrorState] = useState(null);
+
+  // Any current error event object thrown.
```

Comment:
I don't think its obvious what "error event object" refers to here. Do you just mean the `Error` object associated with `errorState` if it is non-null?

Reply:
correct, how about "error object"